src/System.Net.Security/src/System/Net/SecureProtocols/NegoState.Windows.cs

```diff
@@ -13,869 +13,614 @@
+                    }
+                    string authtype = _context.ProtocolName;
 
-        private bool _canRetryAuthentication;
-        private ProtectionLevel _expectedProtectionLevel;
-        private TokenImpersonationLevel _expectedImpersonationLevel;
-        private uint _writeSequenceNumber;
-        private uint _readSequenceNumber;
+                    // TODO #5241: 
+                    // The following call was also specifying WindowsAccountType.Normal, true.
+                    // WindowsIdentity.IsAuthenticated is no longer supported in CoreFX.
+                    result = new WindowsIdentity(token.DangerousGetHandle(), authtype);
+                    return result;
+                }
+                catch (SecurityException)
+                {
+                    // Ignore and construct generic Identity if failed due to security problem.
+                }
+                finally
+                {
+                    if (token != null)
+                    {
+                        token.Dispose();
+                    }
+                }
+            }
 
-        private ExtendedProtectionPolicy _extendedProtectionPolicy;
+            // On the client we don't have access to the remote side identity.
+            result = new GenericIdentity(name, protocol);
+            return result;
+        }
 
-        // SSPI does not send a server ack on successful auth.
-        // This is a state variable used to gracefully handle auth confirmation.
-        private bool _remoteOk = false;
+        internal static string QueryContextAssociatedName(SafeDeleteContext securityContext)
+        {
+            return SSPIWrapper.QueryContextAttributes(GlobalSSPI.SSPIAuth, securityContext, Interop.SspiCli.ContextAttribute.Names) as string;
+        }
 
-        internal NegoState(Stream innerStream, bool leaveStreamOpen)
+        internal static string QueryContextAuthenticationPackage(SafeDeleteContext securityContext)
         {
-            if (innerStream == null)
+            var negotiationInfoClass = SSPIWrapper.QueryContextAttributes(GlobalSSPI.SSPIAuth, securityContext, Interop.SspiCli.ContextAttribute.NegotiationInfo) as NegotiationInfoClass;
+            if (negotiationInfoClass != null)
             {
-                throw new ArgumentNullException("stream");
+                return negotiationInfoClass.AuthenticationPackage;
             }
-
-            _innerStream = innerStream;
-            _leaveStreamOpen = leaveStreamOpen;
+            return null;
         }
 
-        internal static string DefaultPackage
+        internal static object QueryContextSizes(SafeDeleteContext securityContext)
         {
-            get
-            {
-                return NegotiationInfoClass.Negotiate;
-            }
+            return SSPIWrapper.QueryContextAttributes(
+              GlobalSSPI.SSPIAuth,
+              securityContext,
+              Interop.SspiCli.ContextAttribute.Sizes
+              ) as SecSizes;
```

Comment:
If the return type has to stay object (which I hope not) there should be a comment as to why this is doing an `as`.